### ISSUE TITLE
Enum validations throw error\crash

### ISSUE THREAD

**Issue by merof-code** [rails/rails#50332]
Following [a tutor](https://www.shakacode.com/blog/rails-7-1-adds-validation-to-enums/) for new rails 7.1 feature with validation on enum, found that if I use it, add `, validate: true`, it does not work, like at all.

Feature in release notes:
https://edgeguides.rubyonrails.org/7_1_release_notes.html#active-record-notable-changes
- Add validation option for enum.

This is the issue itself
```ruby
class Post < ActiveRecord::Base
  enum status: [:active, :inactive, :deleted], validate: true # does not work
end

def test_association_stuff
    post = Post.new
    post.status = Post.statuses[:active]
    post.save
    assert_equal 1, Post.count
 end
```

## The gist as per template 
[the gist, with rails main](https://gist.github.com/merof-code/28a641979e1b672f95368198903694b8)

### Expected behavior
to not give the error messages

### Actual behavior
```bash
active_record/enum.rb:362:in `assert_valid_enum_definition_values': Enum values true must be either a non-empty hash or an array. (ArgumentError)
```
or 
```bash
You tried to define an enum named "value_type" on the model "MyModel", but this will generate a instance method "Debit?", which is already defined by another enum.
```
### System configuration
latest rails, as in the gist

**Ruby version**: 3.1.4

**Comment by skipkayhil** [comment]
Hey @merof-code, thanks for opening an issue!

The linked article is incorrect, as they are mixing the new and old style enum syntaxes (see 0618d2d84a501aea93c898aec504ff9a0e09d6f2 for the commit that added the new syntax)

This gist passes for me:

```
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  gem "rails", github: "rails/rails", branch: "main"
  gem "sqlite3"
end

require "active_record"
require "minitest/autorun"
require "logger"

# This connection will do for database-independent bug reports.
ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
ActiveRecord::Base.logger = Logger.new(STDOUT)

ActiveRecord::Schema.define do
  create_table :posts, force: true do |t|
    t.integer :status
  end
end

class Post < ActiveRecord::Base
  enum :status, [:active, :inactive, :deleted], validate: true
end

class BugTest < Minitest::Test
  def test_association_stuff
    post = Post.new
    post.status = Post.statuses[:active]
    post.save
    assert_equal 1, Post.count
  end
end
```

Notice that `:status` is a symbol, _not_ a keyword argument. At the moment it appears there is no way to use `validate` with the old syntax (where `status:` is a keyword argument). 

Since there is no bug here, I'm going to close this as we reserve the issue tracker for bugs only, but feel free to reply if you have any question!

**Comment by skipkayhil** [comment]
I've opened #50333 to add support for `_validate` to the `enum status:` style API, but it won't be available until Rails 7.2 at the earliest (unless its omission is considered a bug).

**Comment by merof-code** [comment]
Awesome, I didn't search the docs enough then.
That was unbelievably fast.
